src/java.base/share/classes/java/util/ArrayList.java

```diff
@@ -510,6 +547,36 @@ public E remove(int index) {
         return oldValue;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public E removeFirst() {
+        if (size == 0) {
+            throw new NoSuchElementException();
+        } else {
+            Object[] es = elementData;
+            @SuppressWarnings("unchecked") E oldValue = (E) es[0];
+            fastRemove(es, 0);
+            return oldValue;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public E removeLast() {
+        int sz = size;
+        if (sz == 0) {
+            throw new NoSuchElementException();
+        } else {
+            Object[] es = elementData;
+            sz--;
```

Comment:
This line and the following two lines can be understood as the field size being updated which is not the case.
I think it's more readable to use
```
@SuppressWarnings("unchecked") E oldValue = (E) es[sz - 1];
fastRemove(es, sz - 1);
```
The JIT will common the subexpression 'sz - 1' at runtime.

Reply:
Renamed the variable.